### ISSUE TITLE
[REQUEST] Pennmush full documentation

### ISSUE THREAD

**Issue by mcgiwer** [pennmush/pennmush#1070]
Please include to the project a total documentation of Pennmush, including the one done in Doxygen. This would make the administration easier. Thanks

**Comment by MrWigggles** [comment]
This sounds perfectly reasonable, but I dont think its at all practical. The codebase has been worked, arguably since the 80s. Depending on how you want to count continuity of period mu* codebases that would eventually become Pennmush.

**Comment by mcgiwer** [comment]
If I may suggest, there should be done a base documentation (for example in Doxygen) with would become updated if some major changes were done

**Comment by talvo** [comment]
A lot of the source code is Doxygen commented, though I'm not sure if there's currently a web copy of it which is up to date. However, it won't really do anything for 99% of administrators; unless you're planning to hack the hardcode quite heavily, it's probably of very little interest. (Most minor hacks are in the `*local*.c` files, which are heavily commented for that purpose.)

**Comment by mcgiwer** [comment]
question: is the site: community.penmush.org up to date? it seem that it has same content since year 2007

**Comment by talvo** [comment]
It is, as the name implies, a community-run site, so it's hard to say, though I believe that most of the hardcode-hacking guides and how-tos are still relevant (the main "APIs" in Penn rarely get more than minor changes). Is there something specific you're looking for?

**Comment by mcgiwer** [comment]
yes... nowhere I searched for is described the exact syntax for .db files

**Comment by shawnw** [comment]
http://doxygen.pennmush.org at least is regenerated nightly.

**Comment by mdziczkowski** [comment]
@shawnw It seem that the Doxyfile need to become reconfigured because it's only documenting the source and header files, skiping the other ones with could be documented as well ;-)

**Comment by talvo** [comment]
That is what Doxygen is; it generates documentation for code functions (and constants, structures, etc) based on a particular comment style in the code to provide them.

**Comment by mdziczkowski** [comment]
it depends from the file configuration. I made one with has documented me anything (except some excluded files, likde for example README) and it even has provided a search ;-p
